radio/src/lua/api_general.cpp

```diff
@@ -101,7 +101,7 @@ static constexpr uint8_t maxSourceNameLength{16};
 //
 static Fifo<uint8_t, LUA_FIFO_SIZE>* luaRxFifo = nullptr;
 
-static int luaRxFifoGetByte(void*, uint8_t* data)
+static bool luaRxFifoGetByte(void*, uint8_t* data)
```

Comment:
I understand why you changed this, but this might be an issue when integrating with pure C code, like the USB stack by ST. This is why I was avoiding C++ types in this kind of functions.

Reply:
I totally disagree here ;-)

1) the type of `FIFO::pop()` was already `bool`
2) this is C++ code, not C
3) C has bool since C99 and I don't think it is a good idea to fall back prior to C99
4) the whole problem was introduced due to the wrong implicit conversion from `int` -> `bool`
5) this function was TU-local, so it isn't usable from outside

Reply:
The main issue is that layer separation is not really effectively implemented (yet) here, and some of the callbacks are used in the USB stack. So, yes, these particular functions are not used there, but similar functions are, so that I'd prefer to have a uniform API at the cost of not insisting on C++ conventions. This is the kind of things that needs to dealt with when legacy code is a topic ;-)

Reply:
> C has bool since C99 and I don't think it is a good idea to fall back prior to C99

I'd probably need to check the switches used then, as I remember having troubles with the `gcc` complaining about the `bool` type being not defined.

Reply:
C (>= c99) should / needs to include `stdbool.h`: the C keyword for the built-in type is `_Bool` and `stdbool.h` contains the convenience macros e.g. `bool` and `true` / `false`.
Regarding to `clang` or `gcc` it should be possible `-std=c17` or at least `-std=c11` .